lwt/xen_api_lwt_unix.ml

```diff
@@ -62,29 +62,12 @@ module Lwt_unix_IO = struct
     Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context
 
   let open_connection uri =
-    let domain_addr_t = match Uri.host uri with
-      | Some host ->
-        Lwt.catch
-          (fun () ->
-             Lwt_unix.gethostbyname host >>= fun host_entry ->
-             return (host_entry.Lwt_unix.h_addrtype, host_entry.Lwt_unix.h_addr_list.(0)))
-          (fun _ -> fail (Failed_to_resolve_hostname host))
-      | None -> fail (Failed_to_resolve_hostname "") in
-    (match Uri.scheme uri with
-     | Some "http" -> return false
-     | Some "https" -> return true
-     | Some "file" -> return false
-     | Some x -> fail (Unsupported_scheme x)
-     | None -> fail (Unsupported_scheme "")) >>= fun ssl ->
-    let port = match Uri.port uri with
-      | Some x -> x
-      | None -> if ssl then 443 else 80 in
+    let ssl = Util.is_ssl uri in
```

Comment:
This function may raise an exception where previously it returned a failed promise.

Reply:
oh yes, so `Util.is_ssl uri >>= fun ssl ->`?

Reply:
I don't think that works, you'll need to capture the exception and transform it to a `Lwt.fail ...` and `Lwt.return ssl` otherwise.